File: testdata/zig/sample.zig
const std = @import("std");
const fs = @import("fs");
const json = @import("json");

pub const Config = struct {
    name: []const u8,
    port: u16,
    debug: bool,
};

const InternalState = enum {
    idle,
    running,
    stopped,
};

pub var global_allocator: std.mem.Allocator = undefined;
var internal_count: usize = 0;

pub fn init(allocator: std.mem.Allocator) !void {
    global_allocator = allocator;
}

pub fn parseConfig(data: []const u8) !Config {
    var parser = json.Parser.init(global_allocator);
    defer parser.deinit();
    const result = try parser.parse(data);
    return Config{
        .name = result.get("name").?.string,
        .port = @intCast(result.get("port").?.integer),
        .debug = result.get("debug").?.bool,
    };
}

fn internalHelper(x: i32, y: i32) i32 {
    return x + y;
}

pub fn serve(config: Config) !void {
    var server = std.net.StreamServer.init(.{});
    defer server.deinit();
    try server.listen(std.net.Address.parseIp("0.0.0.0", config.port) catch unreachable);
    while (true) {
        const conn = try server.accept();
        defer conn.stream.close();
        // handle connection
    }
}

File: testdata/nim/sample.nim
import os
import strutils
from json import parseJson, JsonNode
include helpers

type
  Config* = object
    name*: string
    port*: int
    debug: bool

  InternalState = enum
    idle, running, stopped

const VERSION* = "1.0.0"
let defaultPort* = 8080
var requestCount = 0

proc init*(name: string, port: int): Config =
  result = Config(name: name, port: port, debug: false)

proc parseConfig*(data: string): Config =
  let node = parseJson(data)
  result = Config(
    name: node["name"].getStr(),
    port: node["port"].getInt(),
    debug: node["debug"].getBool()
  )

func pureAdd*(x, y: int): int =
  result = x + y

method render*(self: Config): string =
  result = "Config(" & self.name & ":" & $self.port & ")"

proc internalHelper(x: int): int =
  result = x * 2

template debugLog*(msg: string) =
  when defined(debug):
    echo "[DEBUG] " & msg

macro generateField*(name: untyped) =
  discard
